### PR TITLE
Fixed model oom

### PR DESCRIPTION
**Pull request by dangraf**
## What changes are proposed in this pull request?

Added a function "MaxResize" in utils.torch.py and added support for the pipeline to allow an optional parameter "image_max_dim" and "image_max_size" (very similar to image_min_dim and image_min_size". 
This parameter was also added for the model zoo for models with variable image size and was set to 2048 for largest dim of the image.

## How is this patch tested? If it is not, please explain why.

The MaxResize function is tested in "tests/misc/torch_test.py" but no test is added to verify the full pipeline.
Maybe this test should be located in the "unit-test".

### Is this a user-facing change that should be mentioned in the release notes?

-   [ ] No. You can skip the rest of this section.
-   [x] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

### What areas of FiftyOne does this PR affect?

-   [ ] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [x] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

### CODE REVIEW

File: fiftyone/utils/torch.py [hunk 1]
```diff
@@ -114,6 +114,12 @@ class TorchImageModelConfig(foc.Config):
         image_min_dim (None): resize input images during preprocessing, if
             necessary, so that the smaller image dimension is at least this
             value
+        image_max_size (None): resize the input images during preprocessing, if
+            necessary to prevent OOM, so that the image dimensions are at
```
Comment (brimoor) [inline fiftyone/utils/torch.py:118]: nit: I'd recommend removing the "to prevent OOM" commentary from the docstrings in this PR. It's one reason why this parameter might be used, but there could be other reasons

File: fiftyone/utils/torch.py [hunk 2]
```diff
@@ -386,6 +399,11 @@ def _build_transforms(self, config):
         elif config.image_dim:
             transforms.append(torchvision.transforms.Resize(config.image_dim))
 
+        if config.image_max_size:
+            transforms.append(MaxResize(config.image_max_size))
+        elif config.image_max_dim:
+            transforms.append(MinResize(config.image_max_dim))
```
Comment (brimoor) [inline fiftyone/utils/torch.py:405]: `MinResize` -> `MaxResize`

File: tests/misc/torch_tests.py [hunk 3]
```diff
@@ -12,9 +12,47 @@
 import torch
 import torchvision
 
+import numpy as np
+from PIL import Image
+
 import fiftyone as fo
 import fiftyone.utils.torch as fout
 
+def _get_fake_img(h,w):
```
Comment (brimoor) [inline tests/misc/torch_tests.py:21]: Thanks for the tests!

File: fiftyone/utils/torch.py [hunk 4]
```diff
@@ -485,6 +503,43 @@ def __call__(self, pil_image_or_tensor):
         size = (int(round(alpha * h)), int(round(alpha * w)))
         return F.resize(pil_image_or_tensor, size, **self._kwargs)
 
+class MaxResize(object):
```
Comment (brimoor) [inline fiftyone/utils/torch.py:506]: nit: our style uses two newlines between top-level functions/classes. Can you add one more here please?

**Review by brimoor — APPROVED** [review]
LGTM. Thanks for doing this! I left a couple small comments for you; if you could address before merging I'd appreciate it 💯

File: fiftyone/utils/torch.py (hunk 4 above)
Comment (dangraf) [reply fiftyone/utils/torch.py:506]: I've just fixed the issues you addressed. Feels good to give back something since I really like this library. Good work!
